src/NHibernate/Engine/Query/QueryPlanCache.cs

```diff
@@ -24,7 +24,7 @@ public class QueryPlanCache
 		private readonly SimpleMRUCache sqlParamMetadataCache = new SimpleMRUCache();
 
 		// the cache of the actual plans...
-		private readonly SoftLimitMRUCache planCache = new SoftLimitMRUCache(128);
+		private readonly SoftLimitMRUCache<QueryPlanKey, object> planCache = new SoftLimitMRUCache<QueryPlanKey, object>(128);
```

Comment:
Values are different classes here without a common ancestor. I had introduced one first, just to realize it was having almost no added value, because all the cached values currently do not override any `Equals`, causing them to only compare by reference.